tests/static/js/formatters-internal/image.js

```diff
@@ -1,56 +1,106 @@
     });
 
     it('Can restrict the dimensions by height', () => {
-      const imageUrl = Formatters.image(img, 'x338', false).url;
-      expect(imageUrl).toEqual('https://dynm.mktgcdn.com/p/1024x338.jpg');
+      const usImageUrl = Formatters.image(usImg, 'x338', false).url;
+      expect(usImageUrl).toEqual('https://dynm.mktgcdn.com/p/1024x338.jpg');
+      const euImageUrl = Formatters.image(euImg, 'x338', false).url;
+      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=1024,height=338,fit=contain');
     });
 
     it('Can restrict by both dimensions', () => {
-      const imageUrl = Formatters.image(img, '999x338', false).url;
-      expect(imageUrl).toEqual('https://dynm.mktgcdn.com/p/999x338.jpg');
+      const usImageUrl = Formatters.image(usImg, '999x338', false).url;
+      expect(usImageUrl).toEqual('https://dynm.mktgcdn.com/p/999x338.jpg');
+      const euImageUrl = Formatters.image(euImg, '999x338', false).url;
+      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=999,height=338,fit=contain');
     });
 
     it('return the largest image when no dimensions given', () => {
-      const imageUrl = Formatters.image(img, 'x', false).url;
-      expect(imageUrl).toEqual('https://dynm.mktgcdn.com/p/1024x768.jpg');
+      const usImageUrl = Formatters.image(usImg, 'x', false).url;
+      expect(usImageUrl).toEqual('https://dynm.mktgcdn.com/p/1024x768.jpg');
+      const euImageUrl = Formatters.image(euImg, 'x', false).url;
+      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=1024,height=768,fit=contain');
+    });
+  });
+
+  describe('when image is served from EU with no dimensions specified', () => {
+    it('when choosing the biggest image under threshold, use the width and height on the image object if exists', () => {
+      const euImageUrl = Formatters.image(
+        { url: euUrl,
+          width: 1024,
+          height: 768,
+          sourceUrl: 'https://a.mktgcdn.com/p/FOO/516x384.jpg',
+        }, 'x', false).url;
+      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=1024,height=768,fit=contain');
+    });
+
+    it('when choosing the biggest image under threshold, use dimensions from the sourceUrl if width/height does not exist', () => {
+      const euImageUrl = Formatters.image(
+        { url: euUrl,
+          width: 1024,
+          sourceUrl: 'https://a.mktgcdn.com/p/FOO/516x384.jpg'
+        }, 'x', false).url;
+      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=516,height=384,fit=contain');
+    });
+
+    it('when choosing the smallest image over threshold, omit width/height if can\'t parse it from the image object', () => {
+      const euImageUrl = Formatters.image({url: euUrl}, 'x', true).url;
+      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=pad');
```

Comment:
omitting the width/height dimensions with `fit=pad` returns the smallest image? I would have expected it to return the largest image because there are no bounds

Reply:
Taking another look at this. I think `cover` would be a better option. Updated